Add pure PQ ciphersuites with PQ sigs (#4)
rohanmahy/mahy-mls-xwing: draft-mahy-mls-pq.md

and XOF(SHAKE256) (Section 3.2 of {{FIPS202}}) as the Key Derivation Function (KDF).

For the PQ/T hybrid KEMs and the pure ML-KEM HPKE integration, we use the KEMs defined in {{!I-D.ietf-hpke-pq}}.

> The signature scheme for ML-DSA-87 {{MLDSA}} is defined in {{!I-D.ietf-tls-mldsa}}.

Author: ```suggestion
The signature schemes for ML-DSA-65 and ML-DSA-87 {{MLDSA}} are defined in {{!I-D.ietf-tls-mldsa}}.
```

| TBD1 | MLS_128_KitchenSink-KEM(ML-KEM-768,X25519)_AES256GCM_SHA384_Ed25519    |  Y  | RFCXXXX |
| TBD2 | MLS_128_QSF-KEM(ML-KEM-768,P-256)_AES256GCM_SHA384_P256      |  Y  | RFCXXXX |
| TBD3 | MLS_192_QSF-KEM(ML-KEM-1024,P-384)_AES256GCM_SHA384_P384     |  Y  | RFCXXXX |
| TBD4 | MLS_128_ML-KEM-768_AES256GCM_SHA384_P256       |  Y  | RFCXXXX |
| TBD5 | MLS_192_ML-KEM-1024_AES256GCM_SHA384_P384      |  Y  | RFCXXXX |

> | TBD6 | MLS_128_ML-KEM-768_AES256GCM_SHA384_ML-DSA-87  |  Y  | RFCXXXX |

Author: ```suggestion
| TBD6 | MLS_128_ML-KEM-768_AES256GCM_SHA384_ML-DSA-65  |  Y  | RFCXXXX |
```

> | TBD6 | MLS_128_ML-KEM-768_AES256GCM_SHA384_ML-DSA-65  |  Y  | RFCXXXX |

Author: ```suggestion
| TBD6 | MLS_192_ML-KEM-768_AES256GCM_SHA384_ML-DSA-65  |  Y  | RFCXXXX |
```

> | TBD7 | MLS_192_ML-KEM-1024_AES256GCM_SHA384_ML-DSA-87 |  Y  | RFCXXXX |

Author: ```suggestion
| TBD7 | MLS_256_ML-KEM-1024_AES256GCM_SHA512_ML-DSA-87 |  Y  | RFCXXXX |
```

| 0xTBD3 | 0x0052  | 0x0011 | 0x0002 | SHA384 | ecdsa_secp384r1_sha384 |
| 0xTBD4 | 0x0041  | 0x0011 | 0x0002 | SHA384 | ecdsa_secp256r1_sha256 |
| 0xTBD5 | 0x0042  | 0x0011 | 0x0002 | SHA384 | ecdsa_secp384r1_sha384 |
| 0xTBD6 | 0x0041  | 0x0011 | 0x0002 | SHA384 | mldsa87                |
| 0xTBD7 | 0x0042  | 0x0011 | 0x0002 | SHA384 | mldsa87                |

The hash used for the MLS transcript hash is the one referenced in the cipher suite name. "SHA384" refers to the SHA-384 functions defined in [SHS].

# Security Considerations

> This ciphersuites defined in this document combine a post-quantum (or PQ/T hybrid) KEM with a traditional signature algorithm. As such, they are designed to provide confidentiality against quantum and classical attacks, but provide authenticity against classical attacks only.  Thus, these cipher suites do not provide full post-quantum security, only post-quantum confidentiality.
> Cipher suites using post-quantum secure signature algorithms may be defined in the future.

Author: ```suggestion
The first five ciphersuites defined in this document combine a post-quantum (or PQ/T hybrid) KEM with a traditional signature algorithm. As such, they are designed to provide confidentiality against quantum and classical attacks, but provide authenticity against classical attacks only.  Thus, these cipher suites do not provide full post-quantum security, only post-quantum confidentiality.

The last two cipher suites also use post-quantum signature algorithms.
```
